app/src/main/java/org/thoughtcrime/securesms/conversation/newmessage/NewMessageViewModel.kt

```diff
@@ -0,0 +1,94 @@
+package org.thoughtcrime.securesms.conversation.newmessage
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import network.loki.messenger.R
+import nl.komponents.kovenant.ui.failUi
+import nl.komponents.kovenant.ui.successUi
+import org.session.libsession.snode.SnodeAPI
+import org.session.libsignal.utilities.PublicKeyValidation
+import org.thoughtcrime.securesms.ui.GetString
+import javax.inject.Inject
+
+@HiltViewModel
+class NewMessageViewModel @Inject constructor(
+    private val application: Application
+): AndroidViewModel(application), Callbacks {
+
+
+    private val _state = MutableStateFlow(State())
+    val state = _state.asStateFlow()
+
+    private val _event = Channel<Event>()
+    val event = _event.receiveAsFlow()
```

Comment:
For consistency, this seems to have been called `eventFlow` in other classes

Reply:
This is more of a question than a comment. I know that a `Flow` by default can not be re-used, so if we expose a one off Flow to the outside world, does that mean the second time you subscribe it will fail? Maybe we should do this instead?

```suggestion
    val event: Flow<Event> get() = _event.receiveAsFlow()
```

Reply:
I'm not sure I understand. I just meant that you have named the public property for the Channel as `eventFlow` in other classes. It's simply the name, I'm not talking about the Type or anything.

Reply:
Sorry I just hijacked your comment for something unrelated - was talking about potential misuse of `receiveAsFlow`